src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverter.cs

```diff
@@ -0,0 +1,255 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using Corax;
+using Raven.Client;
+using Raven.Server.Json;
+using Raven.Server.Utils;
+using Sparrow;
+using Sparrow.Extensions;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Server;
+using Sparrow.Threading;
+using Voron;
+
+namespace Raven.Server.Documents.Indexes.Persistence.Corax
+{
+    public sealed class CoraxDocumentConverter : ConverterBase, IDisposable
+    {
+        private readonly BlittableJsonTraverser _blittableTraverser;
+        private readonly Dictionary<string, IndexField> _fields;
+        private readonly Index _index;
+        private readonly ByteStringContext _allocator;
+        private readonly Dictionary<Slice, int> _knownFields;
+        private static readonly string _trueLiteral = "true";
+        private static readonly string _falseLiteral = "false";
+
+        public CoraxDocumentConverter(
+                Index index, 
+                bool indexImplicitNull = false, 
+                bool indexEmptyEntries = true, 
+                string keyFieldName = null, 
+                bool storeValue = false, 
+                string storeValueFieldName = Constants.Documents.Indexing.Fields.ReduceKeyValueFieldName)
+        {
+            _index = index ?? throw new ArgumentNullException(nameof(index));
+            _index = index;
+            var fields = index.Definition.IndexFields.Values;
+            var dictionary = new Dictionary<string, IndexField>(fields.Count, default(OrdinalStringStructComparer));
+            foreach (var field in fields)
+                dictionary[field.Name] = field;
+            _fields = dictionary;
+            _blittableTraverser = storeValue ? BlittableJsonTraverser.FlatMapReduceResults : BlittableJsonTraverser.Default;
+            _allocator = new ByteStringContext(SharedMultipleUseFlag.None);
```

Comment:
Don't we have the same assignments in constructor of `LuceneDocumentConverterBase`? I believe we can move most of that to the abstract constructor of `ConverterBase` and have everything in a single place. Then most of the fields e.g. `_index` can be moved there and give them `protected` modifier

Reply:
Moved.

Reply:
Yes, my mistake, moved already.